R/get-sample.R

```diff
@@ -9,3 +9,169 @@ get_samples <- function(con, ...) {
 
   return(assays)
 }
+
+#' Get Samples by Season
+#' @description View sample by season with status and run (if assigned)
+#' @param con connection to the database
+#' @param season year in format YYYY. You can pass in a min and max season as c(YYYY, YYYY)
+#' @param dataset either "raw", "clean", or "unprocessed".
+#' @details the parameter `dataset` can be used to determine what information is
+#' included in the tibble. The `raw`, `clean`, and `unprocessed` datasets contain the following
+#' variables:
+#'
+#' * stream_name
+#' * datetime_collected
+#' * sample_id
+#' * genetic_run_assignment
+#' * field_run_assignment
+#' * fork_length_mm
+#' * fin_clip
+#' * status
+#' * updated_at
+#'
+#' If you select `dataset = "raw"`, the tibble will additionally contain:
+#'
+#' * assay_name
+#' * raw_fluorescence
+#' * threshold
+#' * positive_detection
+#' * plate_run_id
+#'
+#' If you select `dataset = "unprocessed"`, the tibble will additionally contain:
+#'
+#' * assay_name
+#' * sample_type_name
+#' * raw_fluorescence
+#' * background_value
+#' * time
+#' * well_location
+#' * plate_run_id
+#'
+#' @examples
+#' # example database connection
+#' con <- gr_db_connect()
+#' 2022_2023_samples <- get_samples_by_season(con, season = c(2022, 2023), dataset = "clean")
+#' @export
+#' @md
+get_samples_by_season <- function(con, season, dataset = c("raw", "clean", "unprocessed")) {
+
+  if(any(sapply(season, function(i) {nchar(i) == 4})) == FALSE) {
+    stop("You must provide seasons in the format YYYY (i.e. for the 2022 season,
+         please pass the argument as season = 2022; for seasons 2022 and 2023,
+         provide in format c(2022, 2023)")
+  }
+  if(!dataset %in% c("raw", "clean", "unprocessed")) {
```

Comment:
you can use `match.arg` for this since you are providing a vector if choices in the function args, see definition of `update_enum_value` for usage